src/main/java/KeyVaultManagedStorageSamples.java

```diff
@@ -34,72 +31,102 @@ public KeyVaultManagedStorageSamples() throws InterruptedException, ExecutionExc
-        // as well as the storage account.
-
         System.out.println("Creating new vault");
-        Vault vault = azure.vaults().define(VAULT_NAME)
+        vault = azureResourceManager
+                .vaults().define(VAULT_NAME)
                 .withRegion(VAULT_REGION)
                 .withExistingResourceGroup(RESOURCE_GROUP)
                 .defineAccessPolicy()
-                    .forObjectId(USER_OID)
-                    .allowSecretAllPermissions()
-                    .allowStorageAllPermissions()
-                    .attach()
-                .withDeploymentDisabled()
+                .forObjectId(USER_OID)
+                .allowSecretAllPermissions()
+                .allowStorageAllPermissions()
+                .allowCertificateAllPermissions()
+                .allowKeyAllPermissions()
+                .attach()
+                .defineAccessPolicy()
+                .forObjectId(storageAccount.innerModel().identity().principalId())
+                .allowKeyPermissions(KeyPermissions.UNWRAP_KEY,KeyPermissions.WRAP_KEY,KeyPermissions.GET)
+                .attach()
+                .withPurgeProtectionEnabled()
+                .withSoftDeleteEnabled()
                 .create();
         VAULT_URI = vault.vaultUri();
-
+        Key keyVaultKey = vault.keys()
+                .define("key1")
+                .withKeyTypeToCreate(KeyType.RSA)
+                .withKeyCurveName(KeyCurveName.P_256)
+                .withKeyOperations(
+                        KeyOperation.UNWRAP_KEY,
+                        KeyOperation.WRAP_KEY,
+                        KeyOperation.DECRYPT,
+                        KeyOperation.ENCRYPT,
+                        KeyOperation.SIGN,
+                        KeyOperation.VERIFY)
+                .withKeySize(2048)
+                .create();
         System.out.printf("Adding storage account %s to vault %s%n", storageAccount.name(), vault.name());
-
-        StorageAccountAttributes attributes = new StorageAccountAttributes().withEnabled(true);
-        keyVaultClient.setStorageAccount(vault.vaultUri(), storageAccount.name(), storageAccount.id(), "key1", true, "P30D", attributes, null);
+        storageAccount.update().withEncryptionKeyFromKeyVault(VAULT_URI, keyVaultKey.name(), null).apply();
     }
 
     /**
      * Updates a storage account in the vault.
      */
     public void updateStorageAccount() {
-        String keyName = SdkContext.randomResourceName("key", 20);
         // Switch the active key for the storage account
         // Update the key regeneration period
         // Stop rotating the storage account keys
         System.out.println("Updating storage account active key, regeneration period, and disabling automatic key regeneration.");
-        final StorageBundle storageBundle = keyVaultClient.updateStorageAccount(VAULT_URI, STORAGE_ACCOUNT_NAME, "key2", false, "P60D", null, null);
+        Key keyVaultKey = vault.keys()

```

Comment:
Same comment as above about using the management plane library to create a key instead of the data plane library.

Reply:
Yes, this is  using the management plane library to create the second key instead of the data plane library.